### PR TITLE
New resource compose tutorial

### LINKED ISSUE #789: Update Composable Resources Tutorial
### Issue To Be Solved
Need to update the composable resources tutorial to include latest cadence features, best practices, and better instructions.

### Suggest A Solution
@satyamakgec Will post a proposal for an improved tutorial here

### PR DESCRIPTION
**Pull request by ggpt-labs**
Closes https://github.com/onflow/flow/issues/789

### CODE REVIEW

File: tutorials/07-resource-compose/contracts/BestPizzaPlace.cdc [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            self.timeToBake = timeToBake
+            self.sauceType = sauceType
+            self.spiciness = spiciness
+            self.toppings = toppings
+        }
+    }
+
+    init() {
+        self.PizzaStoragePath = /storage/pizza
+        self.PizzaPublicPath = /public/pizza
```
Comment (joshuahannan) [inline tutorials/07-resource-compose/contracts/BestPizzaPlace.cdc:140]: I would create some more specific names for these

**Review by joshuahannan — COMMENTED** [review]
I think this is a great start! You're good to go ahead with working on the tutorial

File: tutorials/07-resource-compose/contracts/BestPizzaPlace.cdc (hunk 1 above)
Comment (gus-labs) [reply tutorials/07-resource-compose/contracts/BestPizzaPlace.cdc]: I changed those now.

**Review by alilloig — COMMENTED** [review]
Great job!!! I love the metaphor for helping people understand about resources. Could it be worth to include two resource interfaces for the Pizza resource, one for exposing `showOrder()` through a public capability and another one for exposing `addTopping()` through a private capability? This way we will be showcasing a little bit about capabilities and you won't have to borrow a reference to the ""whole"" pizza when interacting with it.

Also, since this is a tutorial, I will "fill" all the contract with as much comments as possible to make it super smooth for new comers to understand.

**Comment by alilloig** [comment]
Also if we are thinking of using this tutorial for helping developers to get into the usage of the flow.cli I guess we should include the flow.json file with the contract ready to be deployed?

**Review by satyamakgec — COMMENTED** [review]
It looks good to me .. Need more documentation to do the second pass.

**Comment by gus-labs** [comment]
> Also if we are thinking of using this tutorial for helping developers to get into the usage of the flow.cli I guess we should include the flow.json file with the contract ready to be deployed?

Either that or we describe how to create it, which I did in the voting tutorial.
Which do you prefer?

**Comment by alilloig** [comment]
> > Also if we are thinking of using this tutorial for helping developers to get into the usage of the flow.cli I guess we should include the flow.json file with the contract ready to be deployed?
> 
> Either that or we describe how to create it, which I did in the voting tutorial. What do you prefer?

Provide it already done here and let the users do it by themselves in the next one? (the voting tutorial) That makes sense in my head...

**Comment by gus-labs** [comment]
> Great job!!! I love the metaphor for helping people understand about resources. Could it be worth to include two resource interfaces for the Pizza resource, one for exposing `showOrder()` through a public capability and another one for exposing `addTopping()` through a private capability? This way we will be showcasing a little bit about capabilities and you won't have to borrow a reference to the ""whole"" pizza when interacting with it.
As you suggested, I'm now using those two resource interfaces and expose them via private resp. public capabilities.
`addTopping()` can only be called by the account which stored the pizza, `showOrder()` by everyone.
> Also, since this is a tutorial, I will "fill" all the contract with as much comments as possible to make it super smooth for new comers to understand.
I added lots of comments now.

File: tutorials/07-resource-compose/README.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        // Link capability references
+        acct.link<&AnyResource{BestPizzaPlace.AddTopping}>(BestPizzaPlace.PizzaAddToppingPrivatePath, target: BestPizzaPlace.PizzaStoragePath)
+        acct.link<&AnyResource{BestPizzaPlace.ShowOrder}>(BestPizzaPlace.PizzaShowOrderPublicPath, target: BestPizzaPlace.PizzaStoragePath)
+
+        self.account = acct
+        log("Pizza will be prepared!")
+    }
+
+    post {
+        // TODO: results in failure
```
Comment (franklywatson) [inline tutorials/07-resource-compose/README.md:279]: Why is this todo here? If we plan to keep it maybe need to give more context about why

Comment (gus-labs) [reply tutorials/07-resource-compose/README.md:279]: I'm getting a weird error from the emulator. I'm going to remove this 'post' block for now and create an issue for the error.

**Pull request closed without merging**